transforms/language/text_encoder/python/src/text_encoder_transform.py

```diff
@@ -0,0 +1,137 @@
+        # Make sure that the param name corresponds to the name used in apply_input_params method
+        # of TextEncoderTransform class
+        super().__init__(config)
+        from data_processing.utils import get_logger
+
+        self.logger = get_logger(__name__)
+
+        self.model_name = config.get(model_name_key, default_model_name)
+        self.content_column_name = config.get(
+            content_column_name_key, default_content_column_name
+        )
+        self.output_embeddings_column_name = config.get(
+            output_embeddings_column_name_key, default_output_embeddings_column_name
+        )
+
+        self.model = SentenceTransformer(self.model_name)
+
+    def transform(
+        self, table: pa.Table, file_name: str = None
+    ) -> tuple[list[pa.Table], dict[str, Any]]:
+        """ """
+        self.logger.debug(f"Transforming one table with {len(table)} rows")
+
+        # make sure that the content column exists
+        TransformUtils.validate_columns(
+            table=table, required=[self.content_column_name]
+        )
+
+        embeddings = list(
+            map(
+                lambda x: self.model.encode(x, normalize_embeddings=True),
+                table[self.content_column_name].to_pylist(),
+            ),
+        )
+        result = TransformUtils.add_column(
+            table=table, name=self.output_embeddings_column_name, content=embeddings
+        )
+
+        metadata = {"nfiles": 1, "nrows": len(result)}
+        return [result], metadata
+
+
+class TextEncoderTransformConfiguration(TransformConfiguration):
+    """
+    Provides support for configuring and using the associated Transform class include
+    configuration with CLI args.
+    """
+
+    def __init__(self):
+        super().__init__(
+            name=short_name,
+            transform_class=TextEncoderTransform,
+            # remove_from_metadata=[pwd_key],
+        )
+        from data_processing.utils import get_logger
+
+        self.logger = get_logger(__name__)
+
+    def add_input_params(self, parser: ArgumentParser) -> None:
+        """
+        Add Transform-specific arguments to the given  parser.
+        This will be included in a dictionary used to initialize the TextEncoderTransform.
+        By convention a common prefix should be used for all transform-specific CLI args
+        (e.g, noop_, pii_, etc.)
+        """
+        parser.add_argument(
+            f"--{content_column_name_cli_param}",
+            default=default_content_column_name,
+            help="Column name to get content",
+        )
+        parser.add_argument(
+            f"--{output_embeddings_column_name_cli_param}",
+            default=default_output_embeddings_column_name,
+            help="Column name to store the embeddings",
+        )
+        parser.add_argument(
+            f"--{model_name_cli_param}",
+            default=default_model_name,
+            help="Name of the HF model to use for encoding the text",
```

Comment:
--help from ArgumentParsrer does not seem to show the default values.  Maybe it would be helpful to include the default hf model name in the help text.